metabase/metabase
# truncate total text in pie chart

Closes https://github.com/metabase/metabase/issues/30996

### Description

Prevents the total text in the center of the pie chart from being cut off by automatically truncating it if it's too long.

### How to verify

1. Create a pie chart
2. Use the settings for the measure series to add a very long prefix/suffix to the center text
3. Confirmed that it truncates instead of overflowing
4. In the notebook editor, create another or modify the same question, but use a custom column with a long name (use `concat` function) for the dimension column.
5. On the chart, hover over slices, confirm their labels in the center are truncated.

### Demo

<img width="416" alt="Screenshot 2024-08-02 at 3 10 38 PM" src="https://github.com/user-attachments/assets/599b2439-1203-481f-98c9-9a3ea03c2045">

<img width="413" alt="Screenshot 2024-08-02 at 3 10 30 PM" src="https://github.com/user-attachments/assets/172514a7-9124-4340-896b-3ea978288823">

<img width="421" alt="Screenshot 2024-08-02 at 3 10 12 PM" src="https://github.com/user-attachments/assets/d255bb34-d90f-4178-9ede-802d398d9444">


### Checklist
